tests/unit/model/a_p_i_field_test.py

```diff
@@ -0,0 +1,176 @@
+import pytest
+
+from pyatlan.model.assets import APIField
+from pyatlan.model.enums import APIQueryParamTypeEnum
+from tests.unit.model.constants import (
+    API_CONNECTION_QUALIFIED_NAME,
+    API_CONNECTOR_TYPE,
+    API_FIELD_NAME,
+    API_FIELD_PARENT_OBJECT_QUALIFIED_NAME,
+    API_FIELD_PARENT_QUERY_QUALIFIED_NAME,
+    API_FIELD_REFERENCE_OBJECT_QN,
+)
+
+
+@pytest.mark.parametrize(
+    "name, connection_qualified_name, message",
+    [
+        (None, "connection/name", "connection/name", "name is required"),
+        (
+            API_FIELD_NAME,
+            None,
+            None,
+            "Either parent_api_object_qualified_name or parent_api_query_qualified_name require a valid value",
+        ),
+        (
+            API_FIELD_NAME,
+            "connection/name",
+            "connection/name",
+            "Both parent_api_object_qualified_name and parent_api_query_qualified_name cannot be valid",
+        ),
+    ],
+)
+def test_create_with_missing_parameters_raise_value_error(
+    name: str,
+    parent_api_object_qualified_name: str,
+    parent_api_query_qualified_name: str,
+    message: str,
+):
+    with pytest.raises(ValueError, match=message):
+        APIField.create(
+            name=name,
+            parent_api_object_qualified_name=parent_api_object_qualified_name,
+            parent_api_query_qualified_name=parent_api_query_qualified_name,
+        )
+
+
+def test_create_parent_object():
+    sut = APIField.create(
+        name=API_FIELD_NAME,
+        parent_api_object_qualified_name=API_FIELD_PARENT_OBJECT_QUALIFIED_NAME,
+        parent_api_query_qualified_name=None,
+    )
```

Comment:
We should use `creator()` instead (ref: https://github.com/atlanhq/atlan-python/pull/403/files#r1800760792)